python/mlc_chat/serve/engine.py

```diff
@@ -175,16 +177,29 @@ def __init__(
             ],
         )
         self.trace_recorder = EventTraceRecorder() if enable_tracing else None
-        self._ffi["init"](
-            self.max_single_sequence_length,
-            tokenizer_path,
-            kv_cache_config.asjson(),
-            request_stream_callback,
-            self.trace_recorder,
-            *model_args,
-        )
+        if len(model_args) / 4 > 1:
```

Comment:
Looks somehow confusing, please leave some comments on the argument layout here.

Reply:
Cross link to https://github.com/mlc-ai/mlc-llm/pull/1539/files#r1443033849, I think with EngineMode we can also simplify the if-else logic here.